types.cc

```diff
@@ -3042,6 +3042,48 @@ static bytes serialize_for_cql_aux(const list_type_impl&, collection_mutation_vi
     return collection_type_impl::pack(tmp.begin(), tmp.end(), tmp.size(), sf);
 }
 
+static bytes serialize_for_cql_aux(const user_type_impl& type, collection_mutation_view_description mut, cql_serialization_format) {
+    assert(type.is_multi_cell());
+    assert(mut.cells.size() <= type.size());
+
+    std::vector<bytes> linearized;
+    std::vector<bytes_view_opt> tmp;
+    tmp.reserve(mut.cells.size());
+
+    size_t curr_field_pos = 0;
+    for (auto&& e : mut.cells) {
+        auto field_pos = deserialize_field_index(e.first);
+        assert(field_pos < type.size());
+
+        // Some fields don't have corresponding cells -- these fields are null.
+        while (curr_field_pos < field_pos) {
+            tmp.push_back(std::nullopt);
+            ++curr_field_pos;
+        }
+
```

Comment:
How about size tmp to the ultimate size, then use operator[] to set fields? I wonder how the client deals with trailing nulls due to fields it doesn't know about yet.

Reply:
Done